granadilla/views.py

```diff
@@ -137,6 +143,122 @@ def get_queryset(self):
 contact_list = login_required(ContactListView.as_view())
 
 
+class DeviceCreate(generic_views.CreateView):
+    model = models.LdapDevice
+    template_name = 'granadilla/device.html'
+
+    form_class = LdapDeviceForm
+
+    def get_form_kwargs(self):
+        base_dn = '%s' % (settings.GRANADILLA_DEVICES_DN,)
+        kwargs = super(DeviceCreate, self).get_form_kwargs()
+        kwargs['base_dn'] = base_dn
+        kwargs['username'] = self.request.user.username
+        return kwargs
+
+    def get_success_url(self):
+        return reverse(device_list)
+
+device_create = login_required(DeviceCreate.as_view())
+
+
+class DevicePassword(generic_views.UpdateView):
+    model = models.LdapDevice
+    template_name = 'granadilla/device_password.html'
+
+    form_class = PasswordForm
+
+    def get_object(self, queryset=None):
+        try:
+            device = get_devices().objects.get(device_fullname=\
+                    self.kwargs['device_fullname'])
+        except models.LdapDevice.DoesNotExist:
+            return None
+
+        user = self.request.user
+        if user.username == device.device_username:
+            return device
+        else:
+            return None
+
+    def get_success_url(self):
+        return reverse(device_list)
+
+    def get_form_kwargs(self):
+        base_dn = '%s' % (settings.GRANADILLA_DEVICES_DN,)
+        kwargs = super(DevicePassword, self).get_form_kwargs()
+        kwargs['base_dn'] = base_dn
+        kwargs['device_fullname'] = self.kwargs['device_fullname']
+        return kwargs
+
+    def get_success_url(self):
+        return reverse(device_list)
+
+device_password = login_required(DevicePassword.as_view())
+
+
+class DeviceDelete(generic_views.DeleteView):
+    model = models.LdapDevice
+    template_name = 'granadilla/device_delete.html'
+
+    slug_field = 'device_fullname'
+
+    def get_object(self, queryset=None):
+        try:
+            device = get_devices().objects.get(device_fullname=\
+                    self.kwargs['device_fullname'])
+        except models.LdapDevice.DoesNotExist:
+            return None
+
+        user = self.request.user
+        if user.is_superuser or user.username == device.device_username:
+            return device
+        else:
+            return None
+
+    def get_success_url(self):
+        return reverse(device_list)
+
+
+device_delete = login_required(DeviceDelete.as_view())
+
+
+class DeviceAttrView(generic_views.ListView):
```

Comment:
It lists the characteristics of a device. For the moment there are only the username of the owner and the name of the device.